PBN RCT.Rmd

```diff
@@ -59,114 +61,127 @@ included_participants <- merged_data %>%
 #Make necessary variables numeric
 included_participants <- included_participants %>%
   mutate(across(
-    starts_with(c("b_bhs_", "pi_bhs_", "b_shs_agency_", "pi_shs_agency_", "b_biss_", "pi_biss_", "b_fas_", "pi_fas_", "b_biaaq_", "pi_biaaq_", "b_chime_", "pi_chime_", "pi_pfs_1", "pi_pfs_2", "pi_pfs_3", "pi_pfs_4", "pi_pfs_5", "pi_pfs_6", "pi_pfs_7")),
+    starts_with(c("b_bhs_", "pi_bhs_", "b_shs_agency_", "pi_shs_agency_", "b_biss_", "pi_biss_", "b_fas_", "pi_fas_", "b_biaaq_", "pi_biaaq_", "b_chime_", "pi_chime_", "pi_pfs_1", "pi_pfs_2", "pi_pfs_3", "pi_pfs_4", "pi_pfs_5", "pi_pfs_6", "pi_pfs_7", "b_mfq_", "b_edeq_")),
     as.numeric
   ))
 
-#Create scores and filter for complete cases only
+#Create score and filter for program feedback completers
+included_participants_pfs <- included_participants %>%
+  mutate(pi_pfs_sum = pi_pfs_1 + pi_pfs_2 + pi_pfs_3 + pi_pfs_4 + pi_pfs_5 + pi_pfs_6 + pi_pfs_7) %>%
+  filter(!is.na(pi_pfs_sum))
```

Comment:
This will filter out anybody with _any_ missing PFS values, even if they completed 6/7. Does this happen in our data? If so, what do we want to do with them?

Reply:
Great point! I'm not sure exactly what is going on here, but I've looked into it. vis_miss() seems to indicate that there is one person who did not complete item 1 but completed all of the other items (9% missing for item 1, 8% missing for all the other items). But when I try to compare dataframes to identify who that participant is by their ID, the output I'm getting is character(0) which indicates no difference. This doesn't make sense because the length of one dataframe (check_pfs_partial) is 201 while the length of the other dataframe (check_pfs_all) is 202.
![Screenshot 2024-05-13 at 1 07 25 PM](https://github.com/isaacahuvia/body-neutrality-ssi/assets/116832001/c3414055-8593-47f0-8694-679058806dbb)

Reply:
Try switching the order of the arguments in `setdiff` - I think it returns values in X not in Y

Reply:
That worked thanks! I found the one participant ID where this comes up. How do you think we should handle it?

Reply:
With regards to the PFS specifically, did we pre-register that we would do anything specific with missing values, or limit our statistics to just those who completed the measure? If not, you have the option to take the mean of non-missing items, if you want (there are pro's and con's to this and it's not a big deal either way).

Also, if you are doing anything with this dataset (`included_participants_pfs`) downstream, you just need to keep in mind that you're dropping people who are missing as few as one item on the PFS. That's not a big deal if you're just using the dataset to report descriptive statistics on the PFS, but it could be a problem if you're doing other stuff with the data.